### PR TITLE
fix(ci): Jenkins CI updates

### PR DESCRIPTION
**Pull request by andy-haynes**
This PR modifies the Jenkins CI behavior to work with our non-trunk-based approach using `stable`/`master` and adds branch restrictions for PRs.

### CODE REVIEW

File: Jenkinsfile
Comment (andy-haynes) [reply Jenkinsfile:57]: Yeah, while we're still making the distinction between branches I think that's the way to go. We'd still be deploying after any push to `stable` but we should stick to only merges until we're doing trunk-based development.

So pre-trunk development is:
- dev merges to `master`
- code is deployed to testnet automatically (currently requires manual approval in Jenkins before going live)
- dev merges `master` into `stable`
- code is deployed to mainnet automatically (same manual approval as above)

Post-trunk development is:
- dev merges to `master`
- code is deployed to testnet (probably doesn't need manual approval but we could still require it to encourage testing in the staging env)
- code is deployed to mainnet after manual approval

Comment (morgsmccauley) [reply Jenkinsfile:57]: Cool thanks for explanation 👍🏽

**Review by morgsmccauley — APPROVED** [review]

**Pull request merged**
